traitsui/undo.py

```diff
@@ -27,17 +27,18 @@
 
 import collections
 
+import six
+
 from traits.api import (Event, HasPrivateTraits, HasStrictTraits, HasTraits,
                         Instance, Int, List, Property, Str, Trait)
-import six
-from six.moves import range
+
 
 #-------------------------------------------------------------------------
 #  Constants:
 #-------------------------------------------------------------------------
 
-NumericTypes = (int, int, float, complex)
-SimpleTypes = (str, six.text_type, int, int, float, complex)
+NumericTypes = six.integer_types + (float, complex)
+SimpleTypes = six.string_types + NumericTypes + (bytes,)
```

Comment:
Could replace `six.string_types` with `(six.text_type,)` here to avoid including `str` twice on Python 2.

Reply:
Though I'm not convinced that `bytes` deserves to be called a simple type in Python 3.

Reply:
This is for the purposes of whether to merge undo items - looking at where it is used it comes into play if there is a sequence of things which has been modified, and then the same item of the same sequence gets modified immediately after, then it merges the two.  Not actually sure of the validity of this as a strategy, but it seems if strings are treated this way, so should bytes.